sw/airborne/modules/computer_vision/viewvideo.c

```diff
@@ -109,23 +111,41 @@ struct viewvideo_t viewvideo = {
 #endif
 };
 
+struct image_t img_copy;
+
 /**
  * Handles all the video streaming and saving of the image shots
  * This is a sepereate thread, so it needs to be thread safe!
  */
 struct image_t *viewvideo_function(struct image_t *img);
-struct image_t *viewvideo_function(struct image_t *img)
-{
-  // Resize image if needed
-  struct image_t img_small;
-  image_create(&img_small,
-               img->w / viewvideo.downsize_factor,
-               img->h / viewvideo.downsize_factor,
-               IMAGE_YUV422);
+struct image_t *viewvideo_function(struct image_t *img) {
+  if (!viewvideo.new_image) {
```

Comment:
It also prevents the vision pipeline from copying the image too often if the thread doesn't need it.

I agree that it adds a delay but i'm not sure if there is a better way to get data across threads. Do you know of any?